src/CompanyService.Data/PositionUserRepository.cs

```diff
@@ -26,6 +26,11 @@ public bool Add(DbPositionUser positionUser)
                 throw new ArgumentNullException(nameof(positionUser));
             }
 
+            if (_provider.Positions.Any(p => p.Id == positionUser.PositionId && !p.IsActive))
```

Comment:
You have logic mistake in ChangeUserPositionConsumer.cs you are trying add and delete position. But if it already exists you will have this exception. That is means u will not change position if it exists